File: practice/programmers/js/위장.js
function solution(clothes) {
  const clotheCounter = clothes.reduce((accumulator, [_, category]) => {
    if (accumulator[category]) {
      accumulator[category] += 1;
    } else {
      accumulator[category] = 1;
    }
    return accumulator;
  }, {});
  const product = Object.values(clotheCounter).reduce((accumulator, count) => {
    return accumulator * (count + 1);
  }, 1);
  return product - 1;
}

solution([
  ["yellow_hat", "headgear"],
  ["blue_sunglasses", "eyewear"],
  ["green_turban", "headgear"],
]);

File: practice/programmers/py/위장.py
from collections import defaultdict

def solution(clothes):
    product = 1
    clothe_map = defaultdict(int)
    for _, category in clothes:
        clothe_map[category] += 1
    for clothe_count in clothe_map.values():
        product = product * (clothe_count +1)
    return product - 1

solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], 
["green_turban", "headgear"]])
